src/datastore/postgres-store.ts

```diff
@@ -1722,6 +1722,30 @@ export class PgDataStore
     });
   }
 
+  async getAddressNonceAtBlock(args: {
+    stxAddress: string;
+    blockIdentifier: BlockIdentifier;
+  }): Promise<FoundOrNot<{ nonce: number }>> {
+    return await this.queryTx(async client => {
+      const dbBlock = await this.getBlockInternal(client, args.blockIdentifier);
+      if (!dbBlock.found) {
+        return { found: false };
+      }
+      const executedTxNonce = await client.query<{ nonce: number | null }>(
+        `
+        SELECT MAX(nonce) nonce
+        FROM txs
+        WHERE ((sender_address = $1 AND sponsored = false) OR (sponsor_address = $1 AND sponsored = true))
```

Comment:
I'm curious if we need a new index for this since it's two AND conditions... I realize this is a hotfix to master so fixing here would be impossible if it's required, but we should keep this in mind if we're going to be querying for more sponsored txs in the future.

Reply:
Yep -- this was ripped from the similar query in develop where the index was added. The current indexes make this pretty quick so I'm not concerned. Once develop branch is merged it will be optimized further.